chef/cookbooks/ceph/recipes/conf.rb

```diff
@@ -37,11 +37,43 @@
   keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
 end
 
+if node["ceph"]["config"]["replicas_number"] == 0
+  rep_num = [osd_nodes.length, 3].min
+else
+  rep_num = node["ceph"]["config"]["replicas_number"]
+end
+
+if node["ceph"]["config"]["osds_in_total"] == 0
+  num_osds = osd_nodes.length
+else
+  num_osds = node["ceph"]["config"]["osds_in_total"]
+end
+
+# calculate pg_num based on documentation
+# http://ceph.com/docs/master/rados/operations/placement-groups/
+case num_osds
+when 1..4
+  pg_num = 128
+when 5..9
+  pg_num = 512
+when 10..49
+  pg_num = 4096
+else
+  # Ensure you have a realistic number of placement groups. We recommend
+  # approximately 100 per OSD. E.g., total number of OSDs multiplied by 100
+  # divided by the number of replicas (i.e., osd pool default size).
+  # The result should be rounded up to the nearest power of two.
+  pg_num = 2 ** Math.log2(num_osds * 100 / rep_num).round
```

Comment:
What you've got there based on the upstream docs should be fine (I'm not aware of any better reference)